packages/shared/lib/contexts/governance/utils/isVotingForSelectedProposal.ts

```diff
@@ -3,7 +3,7 @@ import { get } from 'svelte/store'
 import { selectedProposal } from '../stores'
 import { isVotingForProposal } from './isVotingForProposal'
 
-export function isVotingForSelectedProposal(): boolean {
+export async function isVotingForSelectedProposal(accountIndex?: number): Promise<boolean> {
     const proposalId = get(selectedProposal)?.id
-    return isVotingForProposal(proposalId)
+    return isVotingForProposal(proposalId, accountIndex)
```

Comment:
We may want to use this function in a `try/catch` block, so we need to `return await ...` this line, i.e. 
```suggestion
    return await isVotingForProposal(proposalId, accountIndex)
```

See more [here](https://stackoverflow.com/a/42750371/11150203) :+1:

Reply:
Nice StackOverflow link!

What I've understood is that we could return a Promise resolve even after the code reaches the catch block, so where this function is called we could not try-catch it, it would always be resolved, is that it?

If this is for the sole purpose of calling the "isVotingForSelectedProposal" function inside a try-catch block, then I think this change in the code is not necessary

Reply:
The try/catch block doesn't have to be inside the called function, it's a matter of how the caller function uses it. If the caller function uses this function inside its own try/catch block (and it doesn't have the `await`) then it won't trigger the catch, so I do think it is necessary given that we use try/catch blocks often.

Reply:
I made this example to test the try/catch block without return await: https://playcode.io/1066291
You can change the `VALIDATED_STRING` value to test the resolve/reject